cc/toolchain/cc_toolchain_config.bzl

```diff
@@ -0,0 +1,171 @@
+# Copyright (C) 2022 Swift Navigation Inc.
+# Contact: Swift Navigation <dev@swift-nav.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+load(
+    "@bazel_tools//tools/cpp:unix_cc_toolchain_config.bzl",
+    unix_cc_toolchain_config = "cc_toolchain_config",
+)
+
+def cc_toolchain_config(name):
+    # These variables are passed directly through to unix_cc_toolchain_config
+    # below. As far as I can tell they are just metadata that doesn't affect
+    # the build.
+    host_system_name = "linux-x86_64"
+    toolchain_identifier = "clang-x86_64-linux"
+    target_cpu = "k8"
+    target_libc = "glibc_unknown"
+    compiler = "clang"
+    abi_version = "clang"
+    abi_libc_version = "glibc_unknown"
+
+    cxx_builtin_include_directories = [
+        "/include",
+        "/usr/include",
+        "/usr/local/include",
+    ]
+
+    tool_paths = {
+        "ar": "wrappers/llvm-ar",
+        "cpp": "wrappers/clang-cpp",
+        "gcc": "wrappers/clang",
+        "gcov": "wrappers/llvm-profdata",
+        "llvm-cov": "wrappers/llvm-cov",
+        "llvm-profdata": "wrappers/llvm-profdata",
+        "ld": "wrappers/ld.ldd",
+        "nm": "wrappers/llvm-nm",
+        "objcopy": "wrappers/llvm-objcopy",
+        "objdump": "wrappers/llvm-objdump",
+        "strip": "wrappers/llvm-strip",
+    }
+
+    target_system_name = "x86_64-unknown-linux-gnu"
+
+    # Default compiler flags:
+    compile_flags = [
+        "--target=" + target_system_name,
+        # Security
+        "-U_FORTIFY_SOURCE",  # https://github.com/google/sanitizers/issues/247
+        "-fstack-protector",
+        "-fno-omit-frame-pointer",
+        # Diagnostics
+        "-fcolor-diagnostics",
+        "-Wall",
+        "-Wthread-safety",
+        "-Wself-assign",
+    ]
+
+    # -fstandalone-debug disables options that optimize
+    # the size of the debug info.
+    # https://clang.llvm.org/docs/UsersManual.html#cmdoption-fstandalone-debug
+    dbg_compile_flags = ["-g", "-fstandalone-debug"]
+
+    opt_compile_flags = [
```

Comment:
Great that all this complexity is hidden away behind a function with a really simple interface